### PR TITLE
Fixes #30901 - Do not register features when there are pending migrations

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Review by lzap — APPROVED** [review]
Helps to migrate 20200820122057 AddProxySelectorOverrideToRemoteExecutionFeature which was failing for me. I am still having issue with seeding tho.

**Comment by MariaAga** [comment]
Fixed the migrations errors for me and I don't have an issue with db:seed

File: app/models/remote_execution_feature.rb [hunk 1]
```diff
@@ -24,8 +24,10 @@ def self.feature(label)
   end
 
   def self.register(label, name, options = {})
+    pending_migrations = ::Foreman::Plugin.registered_plugins[:foreman_remote_execution]&.pending_migrations
```
Comment (ezr-ondrej) [inline app/models/remote_execution_feature.rb:27]: Just a nitpick, it seems better to use `find` as it feels like a real public interface.
```suggestion
    pending_migrations = ::Foreman::Plugin.find(:foreman_remote_execution)&.pending_migrations
```

**Review by ezr-ondrej — APPROVED** [review]
:+1:

File: app/models/remote_execution_feature.rb (hunk 1 above)
Comment (adamruzicka) [reply app/models/remote_execution_feature.rb:27]: You didn't give me time to actually apply this

Comment (ezr-ondrej) [reply app/models/remote_execution_feature.rb:27]: I didn't think it's important to change now, I don't see the `registred_plugins` going away anytime soon :)
but sorry, will wait for a bit next time ;)

**Pull request merged**